app/src/screens/Dialer/hooks.ts

```diff
@@ -142,18 +141,17 @@ const useDialer = () => {
 
   const activeCall = useActiveCall();
 
-  const isDialerDisabled = React.useMemo(() => {
-    switch (activeCall?.status) {
-      case undefined:
-        return false;
-      case 'fulfilled':
-        return activeCall.callInfo.state !== TwilioCall.State.Disconnected;
-      case 'pending':
-        return true;
-      case 'rejected':
-        return false;
-    }
-  }, [activeCall]);
+  const isDialerDisabled = React.useMemo(
+    () =>
+      match(activeCall)
+        .with({ status: 'pending' }, () => true)
+        .with(
+          { status: 'fulfilled' },
+          (c) => c.info.state !== TwilioCall.State.Disconnected,
+        )
+        .otherwise(() => false),
+    [activeCall],
+  );
```

Comment:
This diff is a good example showing the benefits of pattern matching. Code looks better.